Automattic/wp-calypso
# (wip) Gridicons: use explicit imports

Currently if we use Gridicons we always import the whole package. This PR is the first one of a series of PRs which intent to move imports to be explicit ones. Our goal is to optimize Gridicons usage for code splitting and eventually get rid of the Gridicons package in our build chunk.

This PR does replace all Gridicons which have a simple string identifier `<Gridicon icon="some-icon" />` to decide which Gridicon to use. There several more methods which should be addressed in subsequent PRs.

This
```jsx
import Gridicons from 'gridicons';

<Gridicon icon="arrow-left" size="28" />
```
should become

```jsx
import GridiconArrowLeft from 'gridicons/dist/arrow-left';

<GridiconArrowLeft size="28" />
```

There are two commits (02935e2 and ad32754) which fix eslint warnings and tests which were failing after applying a codemod.

### Testing instructions

I think it's necessary to smoke test Calypso and compare with `master` that all Gridicons are in place. Codewise the above shown pattern should apply to all edited files.

• `import Gridicon from 'gridicons` should be gone and replaced with an explicit import
• All preexisting attributes should still be there e.g. `size={ x }`

Note: there are some files where the original import is still there. That's most likely because there's another `<Gridicon />` in the file which does not use a simple string to define it's `icon`. Those will be addressed in a subsequent PR.
